test/testSaga.test.js

```diff
@@ -182,3 +182,15 @@ test('restarts before done', t => {
       .next().isDone();
   });
 });
+
+test('executes callback passed to yields with actual yielded value', t => {
+  saga
+    .next()
+    .yields((actualValue) => {
+      t.deepEqual(actualValue, take('HELLO'))
+    })
+    .next(action).put({ type: 'ADD', payload: x + y })
+    .next().call(identity, action)
+    .next().fork(otherSaga, z)
+    .next().isDone();
+});
```

Comment:
Can you add some tests cases related to your specific use case? I want to verify that testing an internal saga inside `yields` works. One test with the internal saga passing and one where the internal saga would throw because something is incorrect.